yt_dlp/minicurses.py

```diff
@@ -0,0 +1,162 @@
+import re
+
+from threading import Lock
+from .utils import compat_os_name, get_windows_version
+
+try:
+    # curses is in standard modules, but like sqlite in cookies.py,
+    # it is not present under some circumstances (especially when running on Windows)
+    import curses
+    has_curses = True
+    CURSES_ERROR = (curses.error, )
+except ImportError:
+    has_curses = False
+    CURSES_ERROR = (NameError, )
```

Comment:
isnt it possible to completely get rid of this dependancy by hard coding the UNIX control chars (like for windows)? afaik, all flavors and terminals use the same standard, correct?

Reply:
if we declare that "we only support ANSI escape sequence for Unix platforms", we can hardcode it. then it'll check for `TERM` variable on non-Windows platforms.

Reply:
> Terminal emulators for communicating with local programs as well as remote machines and the text system console almost always support ANSI escape codes. 

https://en.wikipedia.org/wiki/ANSI_escape_code

looks like I can do that

Reply:
done